runner_manager/backend/aws.py

```diff
@@ -31,15 +34,68 @@ def client(self) -> EC2Client:
-        except Exception as e:
-            log.error(e)
-            raise e
+        if self.instance_config.subnet_id and self.instance_config.subnet_configs:
+            raise Exception(
+                "Instance config contains both subnet_id and subnet_configs, only one allowed."
+            )
+        if len(self.instance_config.subnet_configs) > 0:
+            runner = self._create_from_subnet_config(
+                runner, self.instance_config.subnet_configs
+            )
+            log.warn(f"Instance id: {runner.instance_id}")
+        else:
+            instance_resource: AwsInstance = self.instance_config.configure_instance(
+                runner
+            )
+            try:
+                runner = self._create(runner, instance_resource)
+                log.warn(f"Instance id: {runner.instance_id}")
+            except Exception as e:
+                log.error(e)
+                raise e
         return super().create(runner)
 
+    def _create_from_subnet_config(
+        self, runner: Runner, subnet_configs: Sequence[AwsSubnetListConfig]
+    ) -> Runner:
+        # Randomize the order of the Subnets - very coarse load balancing.
+        # TODO: Skip subnets that have failed recently. Maybe with an increasing backoff.
+        order = list(range(len(subnet_configs)))
+        shuffle(order)
+        for idx, i in enumerate(order):
+            subnet_config = subnet_configs[i]
+            try:
+                # Copy the object to avoid modifying the object we were passed.
+                count = self.instance_config.max_count - self.instance_config.min_count
+                log.info(
+                    f"Trying to launch {count} containers on subnet {subnet_config['subnet_id']}"
+                )
+                concrete_instance_config = deepcopy(self.instance_config)
+                concrete_instance_config.subnet_id = subnet_config["subnet_id"]
+                subnet_security_groups = subnet_config.get("security_group_ids", [])
+                if subnet_security_groups:
+                    security_groups = list(concrete_instance_config.security_group_ids)
+                    security_groups += subnet_security_groups
+                    concrete_instance_config.security_group_ids = security_groups
+                instance_resource: AwsInstance = (
+                    concrete_instance_config.configure_instance(runner)
+                )
+                return self._create(runner, instance_resource)
+            except Exception as e:
+                log.warn(
+                    f"Creating instance in subnet {subnet_config['subnet_id']} failed with '{e}'. Retrying with another subnet."
+                )
+                if idx >= len(order) - 1:
+                    raise e
+        return runner
+
+    def _create(self, runner: Runner, instance_resource: AwsInstance) -> Runner:
+        instance = self.client.run_instances(**instance_resource)
```

Comment:
@tcarmet, [the linter](https://github.com/scality/runner-manager/actions/runs/12989983089/job/36239730767?pr=700#step:6:439) is complaining about this line, but it's not a addition,  just moved from the original code. Do you have a preferred way to fix it (I've silenced some warnings before, but I'm not 100% that that's the best plan).

Reply:
Thanks for letting me know, I'm not sure how to fix it as well, but let's merge it as is, I think it's minor and it won't complain on the next lint checks.